src/scheduler/parse.c

```diff
@@ -1161,3 +1181,57 @@ preempt_cmp(const void *p1, const void *p2)
 	}
 }
 
+/**
+ * @brief
+ *	       Frees the memory for the corresponding sort_info structure variable given.
+ *
+ * @param[in]	si_type	- type of sort_info structure
+ *
+ * @return void
+ *
+ */
+void
+free_sort_info(enum sort_info_type si_type)
+{
+	struct sort_info *so_info;
+
+	switch (si_type) {
+		case PRIME_SORT:
+			so_info = conf.prime_sort;
+			break;
+		case NON_PRIME_SORT:
+			so_info = conf.non_prime_sort;
+			break;
+		case PRIME_NODE_SORT:
+			so_info = conf.prime_node_sort;
+			break;
+		case NON_PRIME_NODE_SORT:
+			so_info = conf.non_prime_node_sort;
+			break;
+		default:
+			return;
+	}
+
+	if (so_info != NULL) {
+		int i;
+		for (i = 0; i < MAX_SORTS; i++) {
+			if (so_info[i].res_name != NULL) {
+				free(so_info[i].res_name);
```

Comment:
There is a very long since (like decades old) sort parameter called sort_by/key which used to be how we sorted jobs.  It takes static strings and converts them into the job_sort_key structure.  The code is still there to read sort_by/key.  You should probably remove the parsing code for them so it isn't possible for this to be a static string.

Reply:
Thanks for letting us know this @bhroam. It looks like there are lot of obsolete parameters that we are passing in parse_config() apart from sort_by/key. I feel it would be better if we remove all these obsolete parameters as part of another ticket if we are no longer using them instead of only deleting the parsing of these two parameters.

Reply:
Normally I would agree with you.  The reason I think we should remove sort_by and key now is that they will copy a static string into the res_name member.  You later expect this to be allocated memory so you free it.  If someone uses sort_by/key now, you will free static memory and crash.

Reply:
I am not copying a static string into the res_name member. Please look into line numbers 455 and 460 in parse.c. Instead calling sort_convert[i].res_name and then making use of it in the corresponding sort_info structure variables. So freeing this memory is safe and it does not lead to any crash.

Reply:
I see.  I don't think making a modification to sort_by/key is worthwhile.  I'd rather see it go away.  These have literally been deprecated for 1-2 decades.

Reply:
I have removed the code relating to sort_by/key. Please take a look.